baybe/surrogates/bandit.py

```diff
@@ -0,0 +1,153 @@
+"""Multi-armed bandit surrogates."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, ClassVar
+
+from attrs import define, field
+
+from baybe.exceptions import IncompatibleSearchSpaceError, ModelNotTrainedError
+from baybe.parameters.categorical import CategoricalParameter
+from baybe.parameters.enum import CategoricalEncoding
+from baybe.priors import BetaPrior
+from baybe.searchspace.core import SearchSpace
+from baybe.surrogates.base import Surrogate
+from baybe.targets.binary import _NEGATIVE_VALUE_COMP, _POSITIVE_VALUE_COMP
+from baybe.utils.random import temporary_seed
+
+if TYPE_CHECKING:
+    from botorch.models.model import Model
+    from botorch.posteriors import TorchPosterior
+    from torch import Tensor
+
+
+@define
+class BernoulliMultiArmedBanditSurrogate(Surrogate):
```

Comment:
I must say Im really strongly against having variants of the surrogates coded in their name and would much prefer a general MAB that just does the right logic internally and throws NotImplemented for unsupported variants that are yet to come.

the issue with the other models not accepting binary targets is actually a good point, its just because you didnt update from `NumericalTarget` to `Target` or is there a conceptual problem why a GP wouldnt be able to deal with `y`'s that are either 0 or 1 ? So seems to me they are already compatible with `BinaryTarget`, just a type hint update needed

Conversely, the bandit is not compatible with a numerical target. Internal logic has to be added - so tis a different situation (Im not saying that needs to be fixed in this PR, clearly out of scope).

So I would still prefer a roadmap like this:
- have a generic MAB class, reflected in the name
- it does some logic if target is binary (already present in this PR)
- it does another logic if target is numerical (out of PR scope, so NotImplementedError in this PR)
- it hardcodes beta for binary and normal (or whats suitable) for numerical targets (to be expanded to other dists, out of PR scope)

So the very least you can do is to shape the current code into that so we can avoid deprecations

Reply:
If you are strongly against it, I am willing to change, BUT please note:

* Yes, there is a conceptual problem for the GP. This is not a typing thing but requires an entirely new logic building upon approximate GPs. So even if you don't want to publicly expose something like a "GaussianLikelihoodGaussianProcess" (name just for sake of being explicit), we'll very likely still have this as an isolated class implementation simple because all other variants require a different machinery + different attributes and constructor inputs.
* If I change the BernoulliMAB to a generic MAB, note that the validation needs to check EXACTLY for: 1) BinaryTarget 2) BetaPrior. No other combination is supported right now. So from a user interface, this is a bit frustrating in that you can first incorrectly try with NumericalTargets, get an error, pass a BinaryTarget, get another error because of non-beta prior. If this is what you really want, I can change
* The same argument I explained for the GP also holds for the MAB. The main distinction is basically if things can be solved exactly or require approximations. So for the MAB this means that the implementation can differ vastly depending on whether we use conjugate priors or not. Therefore, also here, we'll probably have at least something like a private "_ExactMAB` class lying around.

Of course, all of that could be hidden behind a dispatcher of some form. So what do you think? Still prefer the validation approach for now? If so, I'll change it, but with the negative side effects mentioned above.

Reply:
re adrians points
the first point just means we cant yet update the type hints in GP
point 2 and 3 just sound like reasonable implementations to me

Reply:
resolved after discussion and agreement